src/execution/vm/bytecode_generator.cpp

```diff
@@ -1449,148 +1467,310 @@ void BytecodeGenerator::VisitBuiltinIndexIteratorCall(ast::CallExpr *call, ast::
       Emitter()->Emit(Bytecode::IndexIteratorFree, iterator);
       break;
     }
-    case ast::Builtin::IndexIteratorGetTinyInt: {
-      LocalVar val = ExecutionResult()->GetOrCreateDestination(ast::BuiltinType::Get(ctx, ast::BuiltinType::Integer));
-      auto col_idx = static_cast<uint16_t>(call->Arguments()[1]->As<ast::LitExpr>()->Int64Val());
-      Emitter()->EmitIndexIteratorGet(Bytecode::IndexIteratorGetTinyInt, val, iterator, col_idx);
+    case ast::Builtin::IndexIteratorGetPR: {
+      ast::Type *pr_type = ast::BuiltinType::Get(ctx, ast::BuiltinType::ProjectedRow);
+      LocalVar pr = ExecutionResult()->GetOrCreateDestination(pr_type);
```

Comment:
The return type of the call should have been set during semantic analysis. So you can simplify to `ExecutionResult()->GetOrCreateDestination(call->Type())`

Reply:
Still valid.